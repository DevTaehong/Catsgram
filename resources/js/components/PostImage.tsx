import React from 'react';
import { createRoot } from 'react-dom/client';

function PostImage() {
    return (
        <div className="card">
            <div>
                <a href="posts/create">
                    <button type="submit" className='create-post'>
                        <span className="create-button-title">Upload your cat!</span>
                    </button>
                </a>
            </div>
        </div>
    );
}

export default PostImage;

if (document.getElementById('postImage')) {
    const container: HTMLElement | null = document.getElementById('postImage');
    if (container) {
        const root = createRoot(container);
        root.render(<PostImage />);
    }
}
