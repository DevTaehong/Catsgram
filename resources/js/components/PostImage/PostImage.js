import React from 'react';
import { createRoot } from 'react-dom/client';
import './PostImage.css';


function PostImage() {
    return (
        <div className="card">
            <div>
                <a href="posts/create">
                    <button type="submit" className='create-post'>
                        <span className="create-button-title">Upload your cat!</span>
                    </button>
                </a>
            </div>
        </div>
    );
}

export default PostImage;

if (document.getElementById('postImage')) {
    const container = document.getElementById('postImage');
    const root = createRoot(container);
    root.render(<PostImage />);
}
