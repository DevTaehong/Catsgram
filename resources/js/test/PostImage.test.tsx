/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer';
import PostImage from '../components/PostImage';
import React from 'react';


it('renders correctly', () => {
    const tree = renderer
    .create(<PostImage />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})
