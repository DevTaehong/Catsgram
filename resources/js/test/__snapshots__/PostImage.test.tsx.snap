// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div>
    <a
      href="posts/create"
    >
      <button
        className="create-post"
        type="submit"
      >
        <span
          className="create-button-title"
        >
          Upload your cat!
        </span>
      </button>
    </a>
  </div>
</div>
`;
